src/Build/BackEnd/Components/RequestBuilder/Lookup.cs

```diff
@@ -658,17 +659,41 @@ internal void AddNewItemsOfItemType(string itemType, ICollection<ProjectItemInst
             IEnumerable<ProjectItemInstance> itemsToAdd = group;
             if (doNotAddDuplicates)
             {
-                // Remove duplicates from the inputs.
-                itemsToAdd = itemsToAdd.Distinct(ProjectItemInstance.EqualityComparer);
 
                 // Ensure we don't also add any that already exist.
                 var existingItems = GetItems(itemType);
+              
+                var existingItemsHashSet = existingItems.ToHashSet(ProjectItemInstance.EqualityComparer);
+
                 if (existingItems.Count > 0)
```

Comment:
I think there's no need to differentiate between the two cases (Count > 0). Let's remove the if and the else block, and just use the body of the if.

Reply:
in one case we are adding both to ExistingItemsHashSet(which started empty due to Count == 0 and to the DeduplicatedItemsToAdd.
in the other case, we are only building one HashSet via the .Distincs call.
Also, we're guaranteed to hit the == 0 case at least once for every item we see so if we care about the memory allocations, I would say that this makes sense to be kept.

Reply:
Actually this only helps if we're not doing logging. Becase when we're doing loggin, we create the HashSet and the .ToList() so the count is same. Only when logging is disabled, this would result in an additional allocation.
I think that this should be fine. Ok, I'll remove the if block.